stormpath/express-stormpath
# Add produces option for /verify

Implements the produces option for /verify according to [the framework spec](https://github.com/stormpath/stormpath-framework-spec/blob/master/email-verification.md#request-handling).

#### How to verify

1. Open up the [Stormpath Console](https://api.stormpath.com/ui2/index.html) and enable the `Verification Email` flow for your directory.
2. Checkout this branch.
3. Clone [express-stormpath-sample-project](https://github.com/stormpath/express-stormpath-sample-project).
4. Use `npm link` and `npm link express-stormpath` to link the library to the example project.
5. Start the server.
6. Navigate to [http://localhost:3000/register](http://localhost:3000/register).
7. Sign up for a new account.
8. Navigate to [http://localhost:3000/verify](http://localhost:3000/verify).
9. Enter the email that you signed up with.
10. Verify that you are redirected to [http://localhost:3000/login?status=unverified](http://localhost:3000/login?status=unverified).
11. Check your email account that you've received a verification link to `http://localhost:3000/verify?sptoken=(unique token)`.
12. Click the link.
13. Verify that you are redirected back to [http://localhost:3000/login?status=verified](http://localhost:3000/login?status=verified).
14. Signup for yet a new account with a new email.
15. Open the terminal and execute `$ http POST localhost:3000/verify login=(your email)@stormpath.com --json`.
16. Verify that you get a `200 OK` response.
17. Modify your configuration and change `web.produces` to `['text/html', '']` (hack because extender currently merges arrays... which needs to be fixed, empty string overwrites second array item.).
18. Verify that step 6-13 works, but that step 15-16 responds with a `404 Not Found` status.
19. Modify the `web.produces` config to `['application/json', '']`.
20. Verify that step 6 results in a `404 Not Found` and that steps 15-16 work as previous.
21. Modify the `web.produces` config again to `['', '']`.
22. Verify that both steps 6 and 15-16 result in a ´404 Not Found`.

Fixes #327.
